tests/handlers/test_presence.py

```diff
@@ -773,6 +804,32 @@ def test_set_presence_from_syncing_keeps_busy(
         # we should still be busy
         self.assertEqual(state.state, PresenceState.BUSY)
 
+    def test_syncing_appropriately_handles_last_active_ts(self) -> None:
+        """Test that syncing does not bump last_active_ts when already online"""
+
+        user_id = "@test:server"
+
+        # Start from a time that is not zero
+        self.reactor.advance(2)
+
+        self._set_presencestate_with_status_msg(user_id, PresenceState.ONLINE, None)
+
+        prev_state = self.get_success(
+            self.presence_handler.get_state(UserID.from_string(user_id))
+        )
+
+        # Poke the reactor so some time passes
+        self.reactor.advance(2)
+        self.get_success(
+            self.presence_handler.user_syncing(user_id, True, PresenceState.ONLINE)
+        )
+
+        state = self.get_success(
+            self.presence_handler.get_state(UserID.from_string(user_id))
+        )
+        # last_active_ts should not have changed as no pro-active event has occurred
+        self.assertEqual(prev_state.last_active_ts, state.last_active_ts)
```

Comment:
@erikjohnston and I went through this and we think there's a misunderstanding here -- `last_active_ts` *should* be updated for each sync. It seems like not updating this except for the initial offline -> online transition is fixing the bug, at the expense of never updating `last_active_ts`.

Reply:
Reading just that summary that sounds like a recipe for disaster.

start: `/sync` (online) => update last_active_ts => new data to send to syncer => goto start

Reply:
> start: `/sync` (online) => update last_active_ts => new data to send to syncer => goto start

My understanding is that the `last_active_ts` should not cause new data to be sent down sync while you're actively syncing.

Reply:
Ah yes

https://spec.matrix.org/v1.7/client-server-api/#last-active-ago

> To reduce the number of presence updates sent to clients the server may include a currently_active boolean field when the presence state is online. When true, the server will not send further updates to the last active time until an update is sent to the client with either a) currently_active set to false or b) a presence state other than online. During this period clients must consider the user to be currently active, irrespective of the last active time.

Reply:
To be clear
> There is more to do in this area, but will be on a different PR as it may be somewhat contentious.

I wasn't done in this area. This doesn't completely fix that `last_active_ts` is being sent inappropriately, just that part that was causing it to be affecting idle mode(I'm trying to stay inside the scope of the point of the PR. The TCP replication was one of the things that was masking and keeping this from working as well, or it would also have been separate).

To fix `last_active_ts` being updated inappropriately, only really means that if `currently_active` is True we don't have to send anything if the only thing to send is an update to `last_active_ts`. The only time `last_active_ts` should be sent is when `currently_active` is False and that update should be changing `currently_active` to True on the same packet. 

State changes are separate from this, and this PR is only supposed to be about making the state change work. I'm sorry this is making problems. I can close this and separate it all out if you would rather. (But I'm not updating the pictures, those were a pain to do the first time)